lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -596,149 +605,221 @@ mod tests {
-                    lat: 45.0,
-                },
-                GeoPoint {
-                    lon: -135.0,
-                    lat: 45.0,
-                },
-                GeoPoint {
-                    lon: -135.0,
-                    lat: -1.0,
-                },
-            ],
-        };
-        let intersect_3 = check_polygon_intersection(&geohash, &polygon_3);
-        assert!(intersect_3);
+        check_intersection(
+            &geohash,
+            build_polygon(
+                vec![
+                    GeoPoint {
+                        lon: -135.0,
+                        lat: -1.0,
+                    },
+                    GeoPoint {
+                        lon: -150.0,
+                        lat: -1.0,
+                    },
+                    GeoPoint {
+                        lon: -150.0,
+                        lat: 46.0,
+                    },
+                    GeoPoint {
+                        lon: -135.0,
+                        lat: 46.0,
+                    },
+                    GeoPoint {
+                        lon: -135.0,
+                        lat: -1.0,
+                    },
+                ],
+                vec![],
+            ),
+            true,
+        );
 
-        // Test a polygon that only share one edge
-        let polygon_4 = GeoPolygon {
-            points: vec![
-                GeoPoint {
-                    lon: -135.0,
-                    lat: -1.0,
-                },
-                GeoPoint {
-                    lon: -150.0,
-                    lat: -1.0,
-                },
-                GeoPoint {
-                    lon: -150.0,
-                    lat: 46.0,
-                },
-                GeoPoint {
-                    lon: -135.0,
-                    lat: 46.0,
-                },
-                GeoPoint {
-                    lon: -135.0,
-                    lat: -1.0,
-                },
-            ],
-        };
-        let intersect_4 = check_polygon_intersection(&geohash, &polygon_4);
-        assert!(intersect_4);
+        check_intersection(
+            &geohash,
+            build_polygon(
+                vec![
+                    GeoPoint {
+                        lon: -134.0,
+                        lat: 1.0,
+                    },
+                    GeoPoint {
+                        lon: -91.0,
+                        lat: 1.0,
+                    },
+                    GeoPoint {
+                        lon: -91.0,
+                        lat: 44.0,
+                    },
+                    GeoPoint {
+                        lon: -134.0,
+                        lat: 44.0,
+                    },
+                    GeoPoint {
+                        lon: -135.0,
+                        lat: -1.0,
+                    },
+                ],
+                vec![],
+            ),
+            true,
+        );
 
-        // Test a geohash that is within the geohash
```

Comment:
I liked those comment explaining the test cases.
Would be great to keep them!

Reply:
added back.